src/tests/blueprint/t_blueprint_mesh_verify.cpp

```diff
@@ -0,0 +1,1163 @@
+//   this list of conditions and the disclaimer below.
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the disclaimer (as noted below) in the
+//   documentation and/or other materials provided with the distribution.
+// 
+// * Neither the name of the LLNS/LLNL nor the names of its contributors may
+//   be used to endorse or promote products derived from this software without
+//   specific prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL SECURITY,
+// LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL 
+// DAMAGES  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, 
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+// POSSIBILITY OF SUCH DAMAGE.
+// 
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+//-----------------------------------------------------------------------------
+///
+/// file: conduit_blueprint_mesh_verify.cpp
+///
+//-----------------------------------------------------------------------------
+
+#include "conduit.hpp"
+#include "blueprint.hpp"
+#include "relay.hpp"
+
+#include <vector>
+#include <string>
+#include "gtest/gtest.h"
+
+using namespace conduit;
+
+/// Helper Functions ///
+
+std::vector<std::string> create_coordsys(const std::string& d1,
+                                         const std::string& d2,
+                                         const std::string& d3="")
+{
+    std::vector<std::string> dim_vector;
+
+    dim_vector.push_back(d1);
+    dim_vector.push_back(d2);
+
+    if(d3 != "")
+    {
+        dim_vector.push_back(d3);
+    }
+
+    return dim_vector;
+}
+
+
+bool is_valid_coordsys(bool (*coordsys_valid_fun)(const Node&, Node&),
+                       const std::vector<std::string>& coordsys)
+{
+    Node n, info;
+
+    bool is_valid = true;
+    for(index_t ci = 0; ci < coordsys.size(); ci++)
+    {
+        const std::string& coordsys_dim = coordsys[ci];
+
+        n[coordsys_dim].set("test");
+        is_valid &= !coordsys_valid_fun(n,info);
+
+        n[coordsys_dim].set(10);
+        is_valid &= coordsys_valid_fun(n,info);
+
+        // FIXME: The coordinate system checking functions shouldn't accept
+        // systems such as (y) or (x, z); all successive dimensions should
+        // require the existence of previous coordsys dimensions.
+        /*
```

Comment:
That makes sense to me!  I'll just leave this code in until we make a final decision on whether or not these types of coordinate spaces will be supported.